app/parse.py

```diff
@@ -13,10 +20,129 @@ class Product:
     price: float
     rating: int
     num_of_reviews: int
+    additional_info: dict
+
+
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+_driver: WebDriver | None = None
+
+
+def get_driver() -> WebDriver:
+    return _driver
+
+
+def set_driver(new_driver: WebDriver) -> None:
+    global _driver
+    _driver = new_driver
+
+
+def product_hdd_block_prices(product_soup: BeautifulSoup) -> dict[str, float]:
+    detail_url = urljoin(BASE_URL, product_soup.select_one(".title")["href"])
+    driver = get_driver()
+    driver.get(detail_url)
+
+    prices = {}
+    try:
+        swatches = driver.find_element(By.CLASS_NAME, "swatches")
+        if swatches:
+            buttons = swatches.find_elements(By.TAG_NAME, "button")
+            for button in buttons:
+                if not button.get_property("disabled"):
+                    button.click()
+                    prices[button.get_property("value")] = float(
+                        driver.find_element(
+                            By.CLASS_NAME, "price"
+                        ).text.replace("$", ""))
+    except NoSuchElementException:
+        return prices
+
+    return prices
+
+
+def parse_single_product(product_soup: BeautifulSoup) -> Product:
+    hdd_prices = product_hdd_block_prices(product_soup)
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(".description").text,
+        price=float(product_soup.select_one(".price").text.replace("$", "")),
+        rating=len(product_soup.select(".ratings span.ws-icon.ws-icon-star")),
+        num_of_reviews=int(
+            product_soup.select_one(".review-count").text.split()[0]
+        ),
+        additional_info={"hdd_prices": hdd_prices}
+    )
+
+
+def get_page_products(url: str, paginate: bool = False) -> [Product]:
+    logging.info(f"Start scraping products from {url}")
+    driver = get_driver()
+    driver.get(url)
+
+    if paginate:
+        try:
+            cookies = driver.find_element(By.CLASS_NAME, "acceptCookies")
+            cookies.click()
+        except ElementNotInteractableException as e:
+            print(e)
+        while True:
+            try:
+                more_button = driver.find_element(
+                    By.CLASS_NAME,
+                    "ecommerce-items-scroll-more"
+                )
+                more_button.click()
+            except Exception as e:
+                print(e)
```

Comment:
the error should be more accurate, try not to use this +  print() is not necessary